actions/src/sema4ai/actions/chat/_client.py

```diff
@@ -156,65 +153,69 @@ def set_bytes(
             # }
             # The client will then do a POST request to the url with the content of the file and
             # should later send the upload complete notification.
-            import json
-            import urllib
-
             import sema4ai_http
 
-            url = f"{self._url}/threads/{thread_id}/files/request-upload"
+            url = f"{self._url}/api/v1/threads/{thread_id}/files/request-upload"
             headers = {
                 "Content-Type": "application/json",
                 "x-action-context": self.get_x_action_context(),
             }
-            data = json.dumps(
-                {"file_name": filename, "file_size": len(content)}
-            ).encode("utf-8")
+            payload = {
+                "file_name": filename,
+                "file_size": len(content)
+            }
 
-            # Send the initial request
-            response = sema4ai_http.post(url, body=data, headers=headers)
+            # Request upload URL from server
+            response = sema4ai_http.post(
+                url,
+                json=payload,
+                headers=headers
+            )
             self._raise_for_status(
                 f"Failed when requesting upload for file {filename} to {self._url}.",
                 response,
                 (200,),
             )
 
             response_data = response.json()
-            file_url = response_data["url"]
+            upload_url = response_data["url"]
             file_id = response_data["file_id"]
             file_ref = response_data["file_ref"]
 
-            parsed_url = urllib.parse.urlparse(file_url)
-            if parsed_url.scheme == "file":
-                # We need to add the file to the local file system
-                p = Path(_uris.to_fs_path(file_url))
-                p.parent.mkdir(parents=True, exist_ok=True)
-                p.write_bytes(content)
+            # If the URL is a local path, write directly to filesystem
+            if upload_url.startswith('/'):
```

Comment:
The old thing was correct here. It should be `file://` to get it to work in the local use case `/`.